### PR TITLE
INFRA-942: Adjust how labels are displayed on CollectionListCard & CollectionInfo

### PR DESCRIPTION
**Pull request by katerinachinnappan**
## Goal

So far, majority of the curators like the option of using a different color scheme to display labels. See [slack thread](https://pocket.slack.com/archives/CTLKW5WCF/p1674679801357179?thread_ts=1674601653.240499&cid=CTLKW5WCF).

* Created a new `ChipLabelsList` component to store the logic for displaying labels wrapped in MUI `Chip` with the expand button. Using a custom style for the MUI Chip `StyledChipComponent`.

* Refactored `CollectionListCard` and `CollectionInfo` to include the new `ChipLabelsList` component.

## Todos

- [x] Testing
- [x] Deploy to dev

## Tickets:

- [https://getpocket.atlassian.net/browse/INFRA-942](https://getpocket.atlassian.net/browse/INFRA-942)

## Implementation Decisions

* Used `event.preventDefault` to prevent the clickable card component firing logic while clicking on the expand label button.

## Demo

https://user-images.githubusercontent.com/16909783/214709120-ff580414-fdcd-4911-93af-6fea1ccbcadb.mov

### CODE REVIEW

File: src/collections/components/ChipLabelsList/ChipLabelsList.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                  label={data?.name}
+                  icon={<LabelOutlinedIcon />}
+                />{' '}
+              </span>
+            );
+          }),
+          <Button
+            key="expand-button"
+            variant="text"
+            onMouseDown={handleMouseDown}
```
Comment (katerinachinnappan) [inline src/collections/components/ChipLabelsList/ChipLabelsList.tsx:66]: can make this optional probably as `CollectionInfo` will suffice with only `onClick` as there is no clickable card on this component.

File: src/collections/components/ChipLabelsList/ChipLabelsList.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+export const ChipLabelsList: React.FC<ChipLabelsListProps> = (props) => {
+  const { collection } = props;
+
+  // expand more/less button state
+  const [expandButtonToggled, setExpandButtonToggle] = useState(true);
+
+  // onMouseDown event is used when pressing the mouse button on the expand button.
+  // this event is needed in order to fire logic as soon as the mouse is pressed, instead of waiting for the full click.
+  const handleMouseDown = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
```
Comment (katerinachinnappan) [inline src/collections/components/ChipLabelsList/ChipLabelsList.tsx:32]: can pass a flag and make this optional as its only needed for the `CollectionListCard` component. `CollectionInfo` does not have a clickable card.

File: src/collections/components/ChipLabelsList/ChipLabelsList.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    setExpandButtonToggle(!expandButtonToggled);
+  };
+
+  // onClick event fires after a full click on a button. This event is also needed
+  // to prevent the clickable collection card component to fire its chain of reactions
+  // having both onMouseDown & onClick events allows to control the button inside a clickable
+  // component
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setExpandButtonToggle(expandButtonToggled);
```
Comment (nina-py) [inline src/collections/components/ChipLabelsList/ChipLabelsList.tsx:48]: ❤️ the detailed comments!

**Review by nina-py — APPROVED** [review]
👍 for comprehensive test coverage!

**Pull request merged**